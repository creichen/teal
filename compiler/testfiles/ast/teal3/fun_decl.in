fun func1 () : RetType;

fun func2 (x, y) : RetType;

fun func3 (x : TypeOfX, y, z : TypeOfZ) : Qualifier [QT] (qf) MyType ;

fun func4 () = {}

fun func5 (x) = if x { return 5; } else { return 6; }