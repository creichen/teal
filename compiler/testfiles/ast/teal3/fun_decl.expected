Module
  List
  List
    FunDecl
      IdDecl Identifier="func1"
      Opt
        Type
          IdUse Identifier="RetType"
          List
          List
      List
      Opt
    FunDecl
      IdDecl Identifier="func2"
      Opt
        Type
          IdUse Identifier="RetType"
          List
          List
      List
        VarDecl
          IdDecl Identifier="x"
          Opt
          Opt
        VarDecl
          IdDecl Identifier="y"
          Opt
          Opt
      Opt
    FunDecl
      IdDecl Identifier="func3"
      Opt
        Type
          IdUse Identifier="MyType"
          List
          List
            TypeQualifier
              IdUse Identifier="Qualifier"
              List
                Type
                  IdUse Identifier="QT"
                  List
                  List
              List
                Access
                  IdUse Identifier="qf"
      List
        VarDecl
          IdDecl Identifier="x"
          Opt
            Type
              IdUse Identifier="TypeOfX"
              List
              List
          Opt
        VarDecl
          IdDecl Identifier="y"
          Opt
          Opt
        VarDecl
          IdDecl Identifier="z"
          Opt
            Type
              IdUse Identifier="TypeOfZ"
              List
              List
          Opt
      Opt
    FunDecl
      IdDecl Identifier="func4"
      Opt
      List
      Opt
        BlockStmt
          List
    FunDecl
      IdDecl Identifier="func5"
      Opt
      List
        VarDecl
          IdDecl Identifier="x"
          Opt
          Opt
      Opt
        IfStmt
          Access
            IdUse Identifier="x"
          BlockStmt
            List
              ReturnStmt
                IntConstant Value="5"
          BlockStmt
            List
              ReturnStmt
                IntConstant Value="6"
